llvm/lib/Target/AIE/AIEMachineScheduler.cpp

```diff
@@ -852,6 +853,23 @@ void AIEPreRASchedStrategy::initialize(ScheduleDAGMI *DAG) {
   }
 }
 
+MachineBasicBlock *AIEPreRASchedStrategy::nextBlock() {
+  MachineBasicBlock *Next = nullptr;
+
+  // The pipeliner is usually disabled to give the postpipeliner a chance.
+  // The prescheduler also clutters the view of the postpipeliner, so we skip
+  // such blocks here.
+  auto Skip = [](MachineBasicBlock *Block) {
```

Comment:
What do you think about a command line option to disable this?

Reply:
We discussed that this is a user-pragma driven affair so that an option is not really necessary. Om second thought I perhaps add one anyway, so that we can easily assess the overall effect of prescheduling